geode-assembly/src/test/java/org/apache/geode/rest/internal/web/RestSecurityIntegrationTest.java

```diff
@@ -62,61 +58,58 @@ public static void before() throws Exception {
   }
 
   @Test
-  public void testFunctions() throws Exception {
-    String json = "{\"@type\":\"double\",\"@value\":210}";
-
-    HttpResponse response = restClient.doGet("/functions", "unknown-user", "1234567");
+  public void testListFunctions() throws Exception {
+    HttpResponse response = restClient.doGet("/functions", "user", "wrongPswd");
     assertEquals(401, GeodeRestClient.getCode(response));
-    response = restClient.doGet("/functions", "stranger", "1234567");
+    response = restClient.doGet("/functions", "user", "user");
     assertEquals(403, GeodeRestClient.getCode(response));
-    response = restClient.doGet("/functions", "dataReader", "1234567");
+    response = restClient.doGet("/functions", "dataRead", "dataRead");
     assertEquals(200, GeodeRestClient.getCode(response));
     response.getEntity();
     assertThat(GeodeRestClient.getContentType(response))
         .containsIgnoringCase(MediaType.APPLICATION_JSON_UTF8_VALUE);
+  }
 
-    response = restClient.doPost("/functions/AddFreeItemsToOrder", "unknown-user", "1234567", json);
+  @Test
+  public void executeFunction() throws Exception {
```

Comment:
A better name for this test would be `executeFunctionNotAuthorized` or something similar to note the the test is only running functions executions that will fail.